### PR TITLE
Minor RPC fixes

### PR DESCRIPTION
**Pull request by codchen**
## Describe your changes and provide context
Fixes for:

[Tony] Invalid value supplied to RpcBlockWithTransactions

[Tony] eth_getTransactionReceipt is missing contractAddress field
curl --location 'https://evm-devnet.seinetwork.io' \
--header 'Content-Type: application/json' \
--data '{"jsonrpc":"2.0","method":"eth_getTransactionReceipt","params":["0x57beed40598914c0d90cd903d883633df9afa8117f5a4bd37ae29a3d670a5218"],"id":1}'

## Testing performed to validate your change
unit tests

### CODE REVIEW

File: evmrpc/tx_test.go [hunk 1]
```diff
@@ -60,6 +60,7 @@ func TestGetTxReceipt(t *testing.T) {
 	require.Equal(t, "0x0123456789012345678902345678901234567890123456789012345678901234", resObj["transactionHash"].(string))
 	require.Equal(t, "0x0", resObj["transactionIndex"].(string))
 	require.Equal(t, "0x1", resObj["type"].(string))
+	require.Equal(t, "0x1234567890123456789012345678901234567890", resObj["contractAddress"].(string))
```
Comment (philipsu522) [inline evmrpc/tx_test.go:63]: nit - can we also add a test case for contractAddress = nil (to prevent potential future regressions)?

**Review by philipsu522 — APPROVED** [review]

File: evmrpc/tx_test.go (hunk 1 above)
Comment (codchen) [reply evmrpc/tx_test.go:63]: yea added

**Comment by philipsu522** [comment]
shipit
